common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -352,6 +352,10 @@ void release(DefaultHandle<T> handle, boolean guarded) {
             }
         }
 
+        private static boolean isTerminated(Thread owner) {
+            return PlatformDependent.isJ9Jvm() ? !owner.isAlive() : owner.getState() == Thread.State.TERMINATED;
```

Comment:
@chrisvest can you add a comment to explain why this is done ?

Reply:
```suggestion
            // Do not use `Thread.getState()` in J9 JVM because it's known to have a performance issue.
            // See: https://github.com/netty/netty/issues/13347#issuecomment-1518537895
            return PlatformDependent.isJ9Jvm() ? !owner.isAlive() : owner.getState() == Thread.State.TERMINATED;
```